design/authn_azure/authn_azure_solution_design.md

```diff
@@ -412,6 +413,47 @@ Azure authenticator performance should conform with our other authenticators wit
 
 - DAP
 
+## Security
+
+### Security boundary
+
+#### Azure security boundary
+The security boundary (who can get an Azure access token) of the Azure Managed Identity Service is the resource to which it is attached to.
+For more details about this boundary, please visit [Azure managed identity security boundary](https://docs.microsoft.com/en-us/azure/active-directory/managed-identities-azure-resources/known-issues#what-is-the-security-boundary-of-managed-identities-for-azure-resources).
+
+#### Conjur security boundary
+The security boundary of Conjur Azure authenticator is the access token received in the authentication request.
+This access token is fetched using Azure Instance Metadata Service (IMDS) which in turn, uses Azure Active Directory (AAD) to generate the token.
+For the full details of how to achieve an access token and how its being generated, please visit [Azure managed identity overview](https://docs.microsoft.com/en-us/azure/active-directory/managed-identities-azure-resources/overview).
+
+
+### Authentication
+Conjur Azure authenticator authenticates the originator by its provided access token.
+The full authentication process described in the [Authentication flow](#authentication-flow) section.
+
+### Authorization
+After authenticating the originator, Conjur Azure authenticator returns a Conjur access token (which is different from the Azure access token).
+This Conjur access token is generated for the Conjur host that was authenticated.
+The authorization will be done according to the following requests and is basically out-of-scope for the Azure authenticator.
+For more details about how Conjur authorization works, please visit [Conjur authorization](https://docs.conjur.org/Latest/en/Content/Operations/Services/authorization.html).
+
+
+ 
+### Azure Access token
+Conjur Azure authenticator receive an Azure access token which is a credential for the Azure resource. As such, Conjur handles this token with the appropriate sensitivity.
+
+#### Azure Access token in transit
+Any request to Conjur done over HTTPS to protect against any kind of Man-In-The-Middle attack. 
+
+#### Azure access token at rest 
+Conjur use that token only for the authentication flow and does not store it or log it into any persistent store.
```

Comment:
uses the Azure access token

Reply:
@shaharglazner  Do we want to say that we log the `xms_mirid` & `oid` fields of the Azure access token?
We also log the Identity Provider URL (e.g `https://sts.windows.net/df242c82-fe4a-47e0-b0f4-e3cb7f8104f1`)

Reply:
@orenbm 
Thanks, good point.
I think that I'm fine with keeping it out because I feel like it's too low-level and will cause confusion, wdyt?

Reply:
I agree, thanks